### PR TITLE
[exporter/Logzioexporter] logs support

### PR DESCRIPTION
**Pull request by yotamloe**
**Description:** <Describe what has changed.>
logzioexporter update:
  * Add support for logs pipeline
  * Remove `logzio-go` and `jaeger-logzio` dependencies
  * Add `logzioSpan`,`logzioService`,`jsonLog` in object directiry
  * Add support for [exporter helper](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/README.md) configuration options (queue, retry)
  * Generic `export` method for traces and logs using HTTP client
  *  Warning logs for soon to be deprecated config options:
       * `custom_endpoint`
       * `drain_interval`
       *  `queue_capacity`
       * `queue_max_length`
  

**Testing:** <Describe what testing was performed and which tests were added.>
Added tests for logs pipeline and for logzio objects in the following files:
* `exporter_test.go`
* `config_test.go`
* `factory_test.go`
* `jsonlog_test.go`
* `logziospan_test.go`

Passing all unit tests:
coverage: 82% of statements
**Documentation:**
* Added documentation for `exporterhelper` configuration option
* Removed deprecated configuration options (custom_endpoint, drain_interval, queue_capacity, queue_max_length)
* Added examples for traces and logs pipeline configuration

### CODE REVIEW

**Comment by dashpole** [comment]
You might want to consider gating the configuration change with a feature-gate.  I did this in the googlecloud exporter when we made changes to our config.  See https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/googlecloudexporter/factory.go#L55 for an example of using different configurations based on a feature-gate.

**Comment by yotamloe** [comment]
Thanks for the advice @dashpole, I'll take a look at that :) 
Regarding the code owners, I'll add the relevant people, should I do it in a different PR?

**Comment by dashpole** [comment]
yes, please add owners in a separate PR.  If you could have one of the other codeowners review this PR, that would be helpful.

**Comment by dashpole** [comment]
Ah, sorry.  I searched for logsio...

**Comment by dashpole** [comment]
You wouldn't need to feature-gate the addition of logs support -- just the removal of config options from your stable configuration.

You just have to abide by [this](https://github.com/open-telemetry/opentelemetry-collector/blob/19cc093b42cb53ec0c7bfd2b9579619e8d89891d/README.md#stable), since your component is stable:

```Breaking changes, including configuration options and the component's output are not expected to happen without prior notice, unless under special circumstances.```

A feature-gate, which is initially disabled by default, is one way to provide prior notice of removal of those options, since users can temporarily (during beta) re-enable them.  But if you have a different way you'd like to provide prior notice (e.g. a warning log when using one of the options you plan to remove), that would also be acceptable.  You should have at least one release where it is in alpha and one in beta, after which you can remove the feature gate.

**Comment by yotamloe** [comment]
Ok, that makes sense.
So if ill keep the legacy options for the next releases, map the values to the new config options when possible (`custom_endpoint` -> `endpoint` for example) and add a warning log for upcoming deprecation and the alternative config options (when the legacy option is used), would that be acceptable?

**Comment by yotamloe** [comment]
Hi @dashpole, I hope you had a nice weekend.
I added the warning messages and option mapping in the latest [commit](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/10821/commits/5258a90faa982db28d94bc34cb32486a4adf8ba3)

File: exporter/logzioexporter/config.go
Comment (dashpole) [inline exporter/logzioexporter/config.go:51]: nit
```suggestion
		logger.Warn("You are using  the deprecated`queue_capacity` option that will be removed in the next release; use exporter helper configuration instead: https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/README.md")
```

File: exporter/logzioexporter/config.go
Comment (dashpole) [inline exporter/logzioexporter/config.go:49]: This function probably doesn't need to be exported.

File: exporter/logzioexporter/config.go
Comment (dashpole) [inline exporter/logzioexporter/config.go:55]:
```suggestion
		logger.Warn("You are using the deprecated `queue_max_length` option that will be removed in the next release; use exporter helper `queue_size` configuration instead: https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/README.md")
```

File: exporter/logzioexporter/config.go
Comment (dashpole) [inline exporter/logzioexporter/config.go:61]:
```suggestion
		logger.Warn("You are using the deprecated `drain_interval` option that will be removed in the next release; use batch processor `timeout` configuration instead: https://github.com/open-telemetry/opentelemetry-collector/tree/main/processor/batchprocessor#batch-processor")
```

File: exporter/logzioexporter/config.go
Comment (dashpole) [inline exporter/logzioexporter/config.go:65]:
```suggestion
		logger.Warn("You are using the deprecated `custom_endpoint` option that will be removed in the next release; please use `endpoint` configuration instead: https://github.com/open-telemetry/opentelemetry-collector/tree/main/config/confighttp")
```

**Review by dashpole — COMMENTED** [review]
I've reviewed the config + readme changes, which look good.  I'll defer to one of the codeowners (who are hopefully more familiar with logzio) to review the log exporter implementation.

**Comment by yotamloe** [comment]
I cant add reviewers. Can you please add @Doron-Bargo as a reviewer?

**Comment by dashpole** [comment]
I can't, since they don't appear to be org members.  Its fine though; they can still review.

**Review by Doron-Bargo — APPROVED** [review]
lgtm

**Comment by yotamloe** [comment]
Hi @dashpole.
Anything else we need to move forward and approve this PR?

File: exporter/logzioexporter/README.md
Comment (dashpole) [inline exporter/logzioexporter/README.md:87]: removing this breaks the link above (I think).  Can you add it back, and also add a link to beta (so the logs link works correctly).

**Review by dashpole — APPROVED** [review]

**Comment by yotamloe** [comment]
Hi @dashpole The lint test is failing due to an [issue](https://github.com/open-telemetry/opentelemetry-collector-contrib/runs/7029019592?check_suite_focus=true) in `tanzuobservabilityexporter` component, `logzioexporter` is passing the tests, how can we resolve this?

**Comment by mx-psi** [comment]
@yotamloe can you fix the merge conflicts?

File: CHANGELOG.md
Comment (mx-psi) [inline CHANGELOG.md:48]: This should be moved to the 'Unreleased' section

File: CHANGELOG.md
Comment (mx-psi) [inline CHANGELOG.md:58]: This should be moved to the 'Unreleased' section

File: exporter/logzioexporter/go.mod
Comment (mx-psi) [inline exporter/logzioexporter/go.mod:85]: These should not be removed, the component still uses both modules

File: exporter/logzioexporter/logzioservice.go
Comment (mx-psi) [inline exporter/logzioexporter/logzioservice.go:34]: Does this need to be public?

File: exporter/logzioexporter/logziospan.go
Comment (mx-psi) [inline exporter/logzioexporter/logziospan.go:58]: Does this need to be public?

File: exporter/logzioexporter/logziospan.go
Comment (mx-psi) [inline exporter/logzioexporter/logziospan.go:31]: Does this need to be public?

File: exporter/logzioexporter/logziospan.go
Comment (mx-psi) [inline exporter/logzioexporter/logziospan.go:27]: Does this need to be public?

File: exporter/logzioexporter/jsonlog.go
Comment (mx-psi) [inline exporter/logzioexporter/jsonlog.go:57]: Does this need to be public?

**Review by mx-psi — COMMENTED** [review]
This PR adds some new public API, is it intentional? In general, functions, methods and structs should be made private so that we can change the internals of the exporter without making breaking changes.

File: exporter/logzioexporter/logzioservice.go
Comment (yotamloe) [reply exporter/logzioexporter/logzioservice.go:34]: I'll make this private

File: exporter/logzioexporter/logziospan.go
Comment (yotamloe) [reply exporter/logzioexporter/logziospan.go:58]: I'll make this private

File: exporter/logzioexporter/logziospan.go
Comment (yotamloe) [reply exporter/logzioexporter/logziospan.go:31]: I'll make this private

File: exporter/logzioexporter/logziospan.go
Comment (yotamloe) [reply exporter/logzioexporter/logziospan.go:27]: I'll make this private

File: exporter/logzioexporter/factory.go
Comment (mx-psi) [inline exporter/logzioexporter/factory.go:109]: Missed these, sorry, if possible we should make them private too

**Review by mx-psi — COMMENTED** [review]
Thanks!! Just a couple more that I missed and we can merge this :)

**Review by mx-psi — APPROVED** [review]

**Pull request merged**
